firmware/Core/Src/rtos_tasks/rtos_tasks.c

```diff
@@ -166,3 +167,52 @@ void TASK_execute_telecommands(void *argument) {
 
 	} /* End Task's Main Loop */
 }
+
+void TASK_monitor_freertos_highstack_watermarks(void *argument) {
+	TASK_HELP_start_of_task();
+
+	while (1) {
+
+		for(uint32_t x = 0; x < task_handles_array_size; x++){
+
+			if(task_handles_array[x].task_handle != NULL)
+			{
+				// Dereferencing the task_handle pointer
+				osThreadId_t dereferenced_task_handle = *(task_handles_array[x].task_handle);
+
+				// Determine the threshold of the task
+				uint32_t task_threshold_words = (task_handles_array[x].stack_size_bytes * CONFIG_highstack_watermark_percentage_threshold) / 400;
```

Comment:
What is magic number 400?

I'd recommend converting to bytes and working with values and percentages in bytes instead of words.

Reply:
I was doing the comparison in words as that was the return value of `uxTaskGetStackHighWaterMark()` hence dividing by a factor of 4. I divided by 100 as the `CONFIG_highstack_watermark_percentage_threshold` is 20 instead of 0.2

Ill move forward in performing the comparison in bytes